KenticoOnboardingApplication/test/KenticoOnboardingApplication.Api.Tests/Controllers/ListControllerTests.cs

```diff
@@ -0,0 +1,297 @@
+                .Returns(itemAndRetrievedItem.retrievedItem);
+
+            var (executedResult, item) =
+                await GetExecutedResultAndValue<Item>(controller =>
+                    controller.GetItemAsync(itemAndRetrievedItem.item.Id));
+
+            Assert.That(executedResult.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+            Assert.That(item, Is.EqualTo(itemAndRetrievedItem.item));
+        }
+
+        [Test]
+        public async Task GetItem_WithEmptyId_ReturnsBadRequest()
+        {
+            var (executedResult, error) =
+                await GetExecutedResultAndValue<HttpError>(controller =>
+                    controller.GetItemAsync(Guid.Empty));
+            var modelStateKeys = error.ModelState.Keys;
+
+            Assert.That(executedResult.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+            Assert.That(modelStateKeys.Contains(nameof(Item.Id)), Is.True);
+            Assert.That(modelStateKeys.Count, Is.EqualTo(1));
+        }
+
+        [Test]
+        public async Task GetItem_WithNonexistingId_ReturnsNotFound()
+        {
+            var id = new Guid(s_ids[2]);
+            _getItemService
+                .GetItemAsync(id)
+                .Returns(RetrievedItem<Item>.Empty);
+
+            var (executedResult, item) =
+                await GetExecutedResultAndValue<Item>(controller => controller.GetItemAsync(id));
+
+            Assert.That(executedResult.StatusCode, Is.EqualTo(HttpStatusCode.NotFound));
+            Assert.That(item, Is.Null);
+        }
+
+        [Test]
+        public async Task PostItem_WithValidItem_ReturnsItemAndLocationAndCreated()
+        {
+            var expectedItem = ItemsCreator.CreateItem(id: s_ids[1], text: s_texts[1]);
+            var postItem = ItemsCreator.CreateItem(text: expectedItem.Text);
+            var expectedLocation = new Uri($"http://localhost/api/{expectedItem.Id}/test");
+            ArrangeUrlLocatorAndCreateItemService(expectedItem, postItem, expectedLocation);
+
+            var (executedResult, item) =
+                await GetExecutedResultAndValue<Item>(controller => controller.PostItemAsync(postItem));
+            var resultLocation = executedResult.Headers.Location;
+
+            Assert.That(executedResult.StatusCode, Is.EqualTo(HttpStatusCode.Created));
+            Assert.That(item, Is.EqualTo(expectedItem));
+            Assert.That(resultLocation, Is.EqualTo(expectedLocation));
+        }
+
+        [Test]
+        [TestCaseSource(nameof(PostTestInvalidItems))]
+        public async Task PostItem_WithInvalidItem_ReturnsBadRequest((Item item, string expectedError) itemWithError)
+        {
+            var (executedResult, error) =
+                await GetExecutedResultAndValue<HttpError>(controller =>
+                    controller.PostItemAsync(itemWithError.item));
+            var modelStateKeys = error.ModelState.Keys;
+
+
+            Assert.That(modelStateKeys.Count, Is.EqualTo(1));
```

Comment:
```suggestion
            Assert.That(error.ModelState, Has.One.Items);
```
[====)-------------](https://imgflip.com/i/2sfajt)